src/libpmemobj/tx.c

```diff
@@ -1234,6 +1253,7 @@ pmemobj_tx_add_common(struct tx *tx, struct tx_range_def *args)
 			 */
 			size_t overlap = rend - MAX(f->offset, r.offset);
 			r.size -= overlap;
+			pmemobj_tx_merge_flags(f, args);
```

Comment:
`f` will be overwritten in 1153. I think this needs to be merged at line 1179 (`pmemobj_tx_merge_flags(fprev, args);`).

Reply:
Initially I also merged there, but I don't think it is necessary because the flag would have already been merged to nprev by one of the other merges. Put another way, if nprev is not NULL, then the the new flag has already been merged into it.